Automattic/sensei
# Split progress models

Resolves #7158 

Splitting models we want to achieve similar behavior for comments-based models and for tables-based models. The most importance difference between them is what statuses we see in the lesson progress. With this PR, both models behave similarly.

Another reason to split models is to make our repositories safer: it's impossible to save a comments-based progress with a tables-based repository.

**An important note here:** we have changes in what status we set in tables-based lesson progress (complete instead of passed), so with enabled tables feature you'll get completed status for the quiz progress in comments and won't be able to grade the quiz. It will be addressed here: https://github.com/Automattic/sensei/issues/7150

## Proposed Changes
* Split progress models into comments-based and tables-based.

## Testing Instructions

0. Create a few courses with lessons and quizzes. Create a student.
1. Tables-based progress not enabled.
2. As a student, take the first course, complete lessons and quizzes.
3. Check there are no errors and you get expected status and behavior both on frontend and in the admin interface. **Make sure with disabled feature flag and reading from tables you can grade submitted quizzes.**
4. Enable tables-based progress.
5. As a student, take the second course, complete lessons and quizzes.
6. Check there are no errors and you get expected status and behavior both on frontend and in the admin interface.
7. Run migrations and enable table-reading.
8. As a student, take the third course, complete lessons and quizzes.
9. Check there are no errors and you get expected status and behavior both on frontend and in the admin interface.

## Pre-Merge Checklist
